File: server/models/Doctor.js

const mongoose = require('mongoose');

const doctorSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true
  },
  specialization: {
    type: String,
    required: true
  },
  qualifications: [String],
  contactNumber: {
    type: String,
    required: true
  },
  email: {
    type: String,
    required: true,
    unique: true
  },
  profilePicture: String,
  dispensaries: [{
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Dispensary'
  }],
  bookingVisibleDays: {
    type: Number,
    default: 30,
    min: 1
  }
}, {
  timestamps: {
    createdAt: true,
    updatedAt: true
  }
});

module.exports = mongoose.model('Doctor', doctorSchema);

File: myclinic-dispensary-app/lib/models/doctor.dart
class Doctor {
  final String id;
  final String name;
  final String? specialization;
  final List<String> qualifications;
  final String? contactNumber;
  final String? email;
  final String? profilePicture;
  final List<String> dispensaryIds;
  final bool disabled;

  Doctor({
    required this.id,
    required this.name,
    this.specialization,
    this.qualifications = const [],
    this.contactNumber,
    this.email,
    this.profilePicture,
    this.dispensaryIds = const [],
    this.disabled = false,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['_id'] ?? json['id'] ?? '',
      name: json['name'] ?? '',
      specialization: json['specialization'],
      qualifications: json['qualifications'] != null
          ? List<String>.from(json['qualifications'])
          : [],
      contactNumber: json['contactNumber'],
      email: json['email'],
      profilePicture: json['profilePicture'],
      dispensaryIds: json['dispensaries'] != null
          ? List<String>.from((json['dispensaries'] as List).map((e) {
              if (e is Map) return e['_id'] ?? e['id'] ?? '';
              return e.toString();
            }))
          : [],
      disabled: json['disabled'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'specialization': specialization,
      'qualifications': qualifications,
      'contactNumber': contactNumber,
      'email': email,
      'profilePicture': profilePicture,
      'dispensaries': dispensaryIds,
    };
  }
}
